src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -238,40 +238,43 @@ void primitive_inst::update_shape() {
     }
 }
 
-void primitive_inst::realloc_if_needed() {
+event::ptr primitive_inst::realloc_if_needed() {
     GPU_DEBUG_GET_INSTANCE(debug_config);
     GPU_DEBUG_PROFILED_STAGE(instrumentation::pipeline_stage::memory_allocation);
 
-
+    event::ptr ev = nullptr;
     // Update param if fake_alignment is available
     auto updated_params = _node->type()->get_fake_aligned_params(*_impl_params);
     auto actual_layout = updated_params.get_output_layout();
     OPENVINO_ASSERT(actual_layout.is_static(), "[GPU] Can't realloc mem for dynamic layout");
 
     // input_layout node is supposed to always use external memory in dynamic case
     if (_node->is_type<input_layout>())
-        return;
+        return ev;
 
     bool can_reuse_buffer = _outputs[0] && actual_layout.count() <= max_output_layout_size;
 
     if (can_reuse_buffer) {
         GPU_DEBUG_TRACE_DETAIL << id() << ": reuse previously allocated output buffer" << std::endl;
         _outputs[0] = _network.get_engine().reinterpret_buffer(*_outputs[0], actual_layout);
+        if (need_reset_output_memory()) {
+            ev = _outputs[0]->fill(_network.get_stream());
```

Comment:
It looks like for USM buffers fill() method has a blocking call, did you try to change this behavior? Maybe it'll improve performance a little

Reply:
I meant forcing of blocking call `const bool blocking = true;` for `gpu_usm::fill()` function

Reply:
I see, I didnt recognized that it was set as blocking call. Now I am curious that why dont we get rid of that blocking true from the fill and  let the sync using the dependencies? hmm...  I will investigate the performance regarding those points in the upcoming works. Thanks for letting me know.

Reply:
@yeonbok, there were some problems related to synchronization, if I remember correctly. But I think we can try to disable it now